runtime/interp.c

```diff
@@ -273,8 +273,9 @@ value caml_interprete(code_t prog, asize_t prog_size)
     value raise_unhandled_closure =
       caml_alloc_1(Closure_tag,
                    Val_bytecode(raise_unhandled_code));
-    raise_unhandled = caml_create_root(raise_unhandled_closure);
-    caml_global_data = caml_create_root(Val_unit);
+    raise_unhandled = raise_unhandled_closure;
+    caml_register_generational_global_root(&raise_unhandled);
+    caml_global_data = Val_unit;
```

Comment:
I don't understand what `caml_global_data` really is. We have places in this PR where it is being altered raw (with the `Field` macro) but we also have other places (for example see `caml_realloc_global` in `meta.c`) where it is being modified with `caml_modify_generational_global_root`. 

If it is a special global root that is operated on raw, then we would need to have something in the root scanning code to operate on it. If it is a generational global root, then we need to register it somewhere and treat it like that everywhere.

Reply:
So `caml_global_data` is treated as a root in multicore and not treated as root in trunk. I mixed up both looking at what trunk does. Hopefully this is fixed now, `caml_global_data` is registered and modified as a generational global root.